datatypes/src/operations/image/to_png.rs

```diff
@@ -0,0 +1,177 @@
+use crate::operations::image::{Colorizer, RgbaTransmutable};
+use crate::raster::{GridPixelAccess, Raster, Raster2D};
+use image::{DynamicImage, ImageFormat, RgbaImage};
+
+pub trait ToPng {
+    /// Outputs png bytes of an image of size width x height
+    fn to_png(&self, width: u32, height: u32, colorizer: &Colorizer) -> Vec<u8>;
+}
+
+impl<T> ToPng for Raster2D<T>
+where
+    T: Into<f64> + Copy + RgbaTransmutable,
+{
+    fn to_png(&self, width: u32, height: u32, colorizer: &Colorizer) -> Vec<u8> {
+        // TODO: use PNG color palette once it is available
+
+        let scale_x = (self.dimension()[0] as f64) / f64::from(width);
+        let scale_y = (self.dimension()[1] as f64) / f64::from(height);
+
+        let color_mapper = colorizer.create_color_mapper();
+
+        let image_buffer: RgbaImage = RgbaImage::from_fn(width, height, |x, y| {
+            // TODO: move these functions to base raster (?)
+            let cell_x = (((f64::from(x) + 0.5) * scale_x) - 0.5).round() as u32;
```

Comment:
you could move them into (static) functions. we can move them to raster if required.

Reply:
0747b67